pkg/providers/openstack/stage.go

```diff
@@ -1,10 +1,201 @@
 package openstack
 
 import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
 	"github.com/emc-advanced-dev/pkg/errors"
+	unikos "github.com/emc-advanced-dev/unik/pkg/os"
 	"github.com/emc-advanced-dev/unik/pkg/types"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/flavors"
+	"github.com/rackspace/gophercloud/openstack/imageservice/v2/images"
+	"github.com/rackspace/gophercloud/pagination"
+	"math"
+	"os"
+	"time"
 )
 
 func (p *OpenstackProvider) Stage(params types.StageImageParams) (_ *types.Image, err error) {
-	return nil, errors.New("not yet supportded for openstack", nil)
+	imageList, err := p.ListImages()
+	if err != nil {
+		return nil, errors.New("failed to retrieve image list", err)
+	}
+
+	// Handle image name collision.
+	for _, image := range imageList {
+		if image.Name == params.Name {
+			if !params.Force {
+				return nil, errors.New(fmt.Sprintf("an image already exists with name '%s', try again with --force", params.Name), nil)
+			} else {
+				logrus.WithField("image", image).Warnf("force: deleting previous image with name '%s'", params.Name)
+				err = p.DeleteImage(image.Id, true)
+				if err != nil {
+					return nil, errors.New("failed to remove existing image", err)
+				}
+			}
+		}
+	}
+
+	clientGlance, err := p.newClientGlance()
+	if err != nil {
+		return nil, err
+	}
+	clientNova, err := p.newClientNova()
+	if err != nil {
+		return nil, err
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"params": params,
+	}).Info("creating boot image from raw image")
+
+	rawImageFile, err := os.Stat(params.RawImage.LocalImagePath)
+	if err != nil {
+		return nil, errors.New("statting raw image file", err)
+	}
+
+	// TODO: Obtain image LOGICAL size, not actual (e.g. 10GB for OSv, not 8MB)
+	imageSizeB := rawImageFile.Size()
```

Comment:
However, we are preparing a new PR with implementation of the dynamic compiler that we were talking about last time.
This compiler will use forked capstan and will be able to set arbitrary image size. I would suggest that after that we modify your OSV_JAVA-COMPILER so that it will use it too.